### ISSUE TITLE
linter: eslint-plugin-react-perf(jsx-no-new-object-as-prop) false positive

### ISSUE THREAD

**Issue by d-tw** [oxc-project/oxc#17743]
### What version of Oxlint are you using?

0.16.12

### What command did you run?

`oxlint --deny-warnings -c ./.oxlintrc.json --tsconfig=./tsconfig.json --type-aware ./src`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
    "$schema": "https://oxc.rs/schema/oxlintrc.json",
    "extends": ["./base.oxlintrc.json"],
    "plugins": ["typescript", "react", "react-hooks", "react-perf"],
    "env": {
        "browser": true,
        "es2024": true
    },
    "rules": {
        "react/jsx-key": "error",
        "react/jsx-no-bind": "error",
        "react/jsx-no-useless-fragment": "error",
        "react/prop-types": "off",
        "react-hooks/rules-of-hooks": "error",
        "react-hooks/exhaustive-deps": [
            "warn",
            {
                "additionalHooks": "(useAsyncCallback|useDebouncedAsyncCallback|useRecoilCallback|useRecoilTransaction_UNSTABLE)"
            }
        ],
        "react-perf/jsx-no-new-array-as-prop": "error",
        "react-perf/jsx-no-new-function-as-prop": "error",
        "react-perf/jsx-no-new-object-as-prop": "error"
    },
    "ignorePatterns": ["node_modules", "dist", "**/data/__generated/**", "**/*.js", "**/*.jsx"]
}
```

With `./base.oxlintrc.json`:

```jsonc
{
    "$schema": "https://oxc.rs/schema/oxlintrc.json",
    "plugins": [
        "typescript"
    ],
    "env": {
        "node": true,
        "es2024": true
    },
    "rules": {
        "no-console": "warn",
        "no-debugger": "warn",
        "no-unused-vars": "off",
        "typescript/no-unused-vars": [
            "warn",
            {
                "varsIgnorePattern": "^_",
                "argsIgnorePattern": "^_",
                "caughtErrorsIgnorePattern": "^_"
            }
        ],
        "typescript/no-floating-promises": "error",
        "typescript/await-thenable": "error",
        "typescript/no-misused-promises": "error",
        "typescript/promise-function-async": "error",
        "typescript/no-deprecated": "error",
        "typescript/restrict-template-expressions": [
            "error",
            {
                "allowNumber": true,
                "allowBoolean": true,
                "allowNullish": true,
                "allowRegExp": true
            }
        ],
        "typescript/no-unsafe-assignment": "off",
        "typescript/no-unsafe-call": "off",
        "typescript/no-unsafe-member-access": "off",
        "typescript/no-unsafe-argument": "off",
        "typescript/no-unsafe-return": "off",
        "typescript/require-await": "off"
    },
    "ignorePatterns": [
        "node_modules",
        "dist",
        "**/*.js",
        "**/*.jsx"
    ]
}
```

### What happened?

There's a false positive for the `eslint-plugin-react-perf(jsx-no-new-object-as-prop)` rule.

Whilst a new object is created during each render, the type of the value that is ultimately passed as a prop is a scalar (a `number | undefined` in this case), so it won't ever trigger an unwanted re-render. 

This code passes when using eslint.

Simplified source:

```tsx
export type FloatBounds = {
    max?: number
    min?: number
}

type FloatBoundsInputProps = {
    value?: FloatBounds
}

export const FloatBoundsInput: FC<FloatBoundsInputProps> = ({ value }) => {
    const { min } = value ?? {}

    return <InputNumber value={min} />
}
```

Error
```
  × eslint-plugin-react-perf(jsx-no-new-object-as-prop): JSX attribute values should not contain objects created in the same scope.
    ╭─[./FloatBoundsInput.tsx:36:11]
 35 │ export const FloatBoundsInput: FC<FloatBoundsInputProps> = ({ value }) => {
 36 │     const { min } = value ?? {}
    ·           ───┬───            ─┬
    ·              │                ╰── And assigned a new value here
    ·              ╰── The prop was declared here
 37 │ 
 38 │     return <InputNumber value={min} />
    ·                                ─┬─
    ·                                 ╰── And used here
 39 │ }
    ╰────
```

**Comment by camc314** [comment]
Wouldn't it be better to write this as
```ts
const min = value.min
```

That way you can avoid the new object in the case where `value` is nullish?

I'm tempted to say this is expected behaviour, as, in the case where `value` is nullish, it's creating a new object on every rerender
